checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1602 @@
+        typeFactory.createCalledMethods(mustCallValues.toArray(new String[0]));
+    return typeFactory.getQualifierHierarchy().isSubtype(cmAnno, cmAnnoForMustCallMethods);
+  }
+
+  /**
+   * Is {@code exceptionClassName} an exception type we are ignoring, to avoid excessive false
+   * positives? For now we ignore {@code java.lang.Throwable}, {@code NullPointerException}, and the
+   * runtime exceptions that can occur at any point during the program due to something going wrong
+   * in the JVM, like OutOfMemoryErrors or ClassCircularityErrors.
+   *
+   * @param exceptionClassName the fully-qualified name of the exception
+   * @return true if the given exception class should be ignored
+   */
+  private static boolean isIgnoredExceptionType(@FullyQualifiedName Name exceptionClassName) {
+    // any method call has a CFG edge for Throwable/RuntimeException/Error to represent run-time
+    // misbehavior. Ignore it.
+    return exceptionClassName.contentEquals(Throwable.class.getCanonicalName())
+        || exceptionClassName.contentEquals(RuntimeException.class.getCanonicalName())
+        || exceptionClassName.contentEquals(Error.class.getCanonicalName())
+        // use the Nullness Checker to prove this won't happen
+        || exceptionClassName.contentEquals(NullPointerException.class.getCanonicalName())
+        // these errors can't be predicted statically, so we'll ignore them and assume they won't
+        // happen
+        || exceptionClassName.contentEquals(ClassCircularityError.class.getCanonicalName())
+        || exceptionClassName.contentEquals(ClassFormatError.class.getCanonicalName())
+        || exceptionClassName.contentEquals(NoClassDefFoundError.class.getCanonicalName())
+        || exceptionClassName.contentEquals(OutOfMemoryError.class.getCanonicalName())
+        // it's not our problem if the Java type system is wrong
+        || exceptionClassName.contentEquals(ClassCastException.class.getCanonicalName())
+        // it's not our problem if the code is going to divide by zero.
+        || exceptionClassName.contentEquals(ArithmeticException.class.getCanonicalName())
+        // use the Index Checker to catch the next two cases
+        || exceptionClassName.contentEquals(ArrayIndexOutOfBoundsException.class.getCanonicalName())
+        || exceptionClassName.contentEquals(NegativeArraySizeException.class.getCanonicalName())
+        // Most of the time, this exception is infeasible, as the charset used
+        // is guaranteed to be present by the Java spec (e.g., "UTF-8"). Eventually,
+        // we could refine this exclusion by looking at the charset being requested
+        || exceptionClassName.contentEquals(UnsupportedEncodingException.class.getCanonicalName());
+  }
+
+  /**
+   * Updates {@code visited} and {@code worklist} if the input {@code state} has not been visited
+   * yet.
+   *
+   * @param state the current state
+   * @param visited the previously-analyzed states
```

Comment:
`visited` is not the previously-analyzed states.  It is states that have been placed on the worklist already.  They might not have been analyzed yet.  Please update documentation throughout.